python/ray/data/tests/test_pandas_block.py

```diff
@@ -48,7 +54,203 @@ def fn2(batch):
     assert isinstance(block, pd.DataFrame)
 
 
-if __name__ == "__main__":
-    import sys
+class TestSizeBytes:
+    def test_size_bytes_small(ray_start_regular_shared):
+        animals = ["Flamingo", "Centipede"]
+        block = pd.DataFrame({"animals": animals})
+
+        block_accessor = PandasBlockAccessor.for_block(block)
+        bytes_size = block_accessor.size_bytes()
+
+        # check that memory usage is within 10% of the size_bytes
+        # For strings, Pandas seems to be fairly accurate, so let's use that.
+        memory_usage = block.memory_usage(index=True, deep=True).sum()
+        assert memory_usage * 0.9 <= bytes_size <= memory_usage * 1.1, (
+            bytes_size,
+            memory_usage,
+        )
+
+    def test_size_bytes_large_str(ray_start_regular_shared):
+        animals = [
+            random.choice(["alligator", "crocodile", "centipede", "flamingo"])
+            for i in range(100_000)
+        ]
+        block = pd.DataFrame({"animals": animals})
+        # if you remove this line it breaks
+        block["animals"] = block["animals"].astype("string")
```

Comment:
@Bye-legumes could you take a look at this?

Reply:
For this test, if we use ` block["animals"] = block["animals"].astype("string")`, the detype is `animals    string[python]
dtype: object` and we can use `memory_usage = block.memory_usage(index=True, deep=True).sum()` to get the correct memory.
if we do not use 
` block["animals"] = block["animals"].astype("string")`, it will be the object `animals    object
dtype: object
5775025`, and  `memory_usage = block.memory_usage(index=True, deep=True).sum()` cannot give the correct usage as this issue motioned before and the correct bytes is 58*100_000 in real. the real size is the size of string in list + list ref size `memory_usage = sys.getsizeof(animals[0])*100_000
`